admin-dev/themes/new-theme/js/pages/product/components/combination-list/bulk-choices-selector.ts

```diff
@@ -40,16 +41,43 @@ export default class BulkChoicesSelector {
 
   private tabContainer: HTMLDivElement;
 
-  constructor(eventEmitter: EventEmitter, tabContainer: HTMLDivElement) {
+  private paginatedCombinationsService: PaginatedCombinationsService
+
+  constructor(
+    eventEmitter: EventEmitter,
+    tabContainer: HTMLDivElement,
+    paginatedCombinationsService: PaginatedCombinationsService,
+  ) {
     this.eventEmitter = eventEmitter;
     this.tabContainer = tabContainer;
+    this.paginatedCombinationsService = paginatedCombinationsService;
     this.init();
   }
 
+  //@todo: may become private or unused
   public getSelectedCheckboxes(): NodeListOf<HTMLInputElement> {
     return this.tabContainer.querySelectorAll<HTMLInputElement>(`${CombinationMap.tableRow.isSelectedCombination}:checked`);
   }
 
+  //@todo: could as well return count?
+  public async getSelectedIds(): Promise<number[]> {
+    const allSelected = this.tabContainer.querySelector<HTMLInputElement>(`${CombinationMap.bulkSelectAll}:checked`);
+
+    if (allSelected) {
+      //@todo: hardcoded productId
```

Comment:
I though the productId was already injected in the `paginatedCombinationsService` ?

Reply:
Also the filters are already injected in the service right? It means you don't need to care about the difference between "Select all" and "Select filtered" right? 

If so that's convenient ^^ But maybe we should a comment here to explain it, since it's not intuitive when you read the code for the first time